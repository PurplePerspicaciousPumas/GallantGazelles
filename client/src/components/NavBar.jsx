import React, { Component } from 'react';
import LoggedInNav from './LoggedInNav.jsx';
import LoggedOutNav from './LoggedOutNav.jsx';
import { Grid, Header, Icon, Input, Menu } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { checkSession } from '../actions/user';

class NavBar extends Component {
  constructor(props) {
    super(props)
  }
  componentWillMount() {
    const {dispatch, getUserInfo} = this.props;
    getUserInfo();
  }
  render () {
    if (this.props.username) {
      return <LoggedInNav username={this.props.username}/>
    } else {
      return <LoggedOutNav />
    }
  }
}

const mapStateToProps = (state) => {
  return {
    username: state.user.username
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getUserInfo: () => { dispatch(checkSession()) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);