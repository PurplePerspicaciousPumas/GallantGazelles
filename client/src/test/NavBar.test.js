import React from 'react';
import NavBar from '../components/NavBar.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <NavBar />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});