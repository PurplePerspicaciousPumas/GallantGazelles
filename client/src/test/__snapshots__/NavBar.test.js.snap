// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <div>
    <span>
      Home 
    </span>
    <span>
      Start a Pitch 
    </span>
    <span>
      How it works 
    </span>
    <span>
      Companies 
    </span>
    <span>
      Sign in 
    </span>
    <span>
      Sign up 
    </span>
  </div>
  <div>
    <input
      name="search"
      placeholder="Search"
      type="text"
    />
  </div>
</nav>
`;
